x-pack/plugins/cases/server/common/limiter_checker/limiters/files.ts

```diff
@@ -5,24 +5,34 @@
  * 2.0.
  */
 
-import { buildFilter } from '../../../client/utils';
-import { CommentType, FILE_ATTACHMENT_TYPE } from '../../../../common/api';
+import type { FileServiceStart } from '@kbn/files-plugin/server';
+import { CommentType } from '../../../../common/api';
 import type { CommentRequest } from '../../../../common/api';
-import { CASE_COMMENT_SAVED_OBJECT, MAX_FILES_PER_CASE } from '../../../../common/constants';
+import { MAX_FILES_PER_CASE } from '../../../../common/constants';
 import { isFileAttachmentRequest } from '../../utils';
 import { BaseLimiter } from '../base_limiter';
 
 export class FileLimiter extends BaseLimiter {
-  constructor() {
+  constructor(private readonly fileService: FileServiceStart) {
     super({
       limit: MAX_FILES_PER_CASE,
       attachmentType: CommentType.externalReference,
       field: 'externalReferenceAttachmentTypeId',
-      filter: createFileFilter(),
       attachmentNoun: 'files',
     });
   }
 
+  public async countOfItemsWithinCase(caseId: string): Promise<number> {
+    const files = await this.fileService.find({
+      perPage: 1,
+      meta: {
+        caseIds: caseId,
```

Comment:
This should be an array right? `caseIds: [caseId]`

Reply:
Oops, yeah. The reason it works is because the file service `find` converts everything to an array internally. I'll change it 👍